include/simdjson/generic/ondemand/value-inl.h

```diff
@@ -95,6 +95,15 @@ simdjson_really_inline simdjson_result<array_iterator> value::begin() & noexcept
 simdjson_really_inline simdjson_result<array_iterator> value::end() & noexcept {
   return {};
 }
+simdjson_really_inline simdjson_result<size_t> value::count_elements() & noexcept {
+  simdjson_result<size_t> answer;
+  auto a = get_array();
+  answer = a.count_elements();
+  // count_elements leaves you pointing inside the array, at the first element.
+  // We need to move back.
+  iter.move_at_start();
```

Comment:
Is it illegal for the position to be at the first value of the array when you start iteration now? I know you changed that bit, just checking. Before, we allowed the cursor to move past the `[` when you started an array, and checked for it (to see if it was the first element or not).

Long as that's the case, I'm all :+1: on this.

Reply:
> Is it illegal for the position to be at the first value of the array when you start iteration now? I know you changed that bit, just checking.  

I do not think I did change that. I don't recall touching the iteration at all. I think that to be able to iterate, you have to be right at the `[` otherwise it fails. I think it explicitly checks for `[`.

Reply:
Oh, to open a new array you probably do need to be at a [. Ok.